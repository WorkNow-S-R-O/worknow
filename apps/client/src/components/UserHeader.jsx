/* eslint-disable no-unused-vars */
import React from 'react';
import PropTypes from 'prop-types'

 const UserHeader = ({ user }) => (
  <div className="d-flex flex-column align-items-center mb-4">
    <img
      src={user.imageUrl || "/images/default-avatar.png"}
      alt="User Avatar"
      className="rounded-circle mb-3"
      style={{ width: "100px", height: "100px", objectFit: "cover" }}
    />
    <div>
      <h2>{user.firstName ? `${user.firstName} ${user.lastName || ""}` : "Анонимный пользователь"}</h2>
      <p className="text-muted">{user.email || "Email не указан"}</p>
    </div>
  </div>
);



UserHeader.propTypes = {
  user: PropTypes.shape({
    imageUrl: PropTypes.string,
    firstName: PropTypes.string,
    lastName: PropTypes.string,
    email: PropTypes.string,
  }).isRequired,
};

export default UserHeader;