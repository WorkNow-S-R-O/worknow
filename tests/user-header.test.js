/* eslint-disable no-unused-vars */
/* global test, expect, jest */
import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import UserHeader from '../apps/client/src/components/UserHeader';

const baseProps = {
  loading: false,
  setImageError: jest.fn(),
};

const mockUser = {
  firstName: 'John',
  lastName: 'Doe',
  email: 'john.doe@example.com',
  imageUrl: '',
};

test('renders UserHeader correctly', () => {
  const { getByText, getByAltText, asFragment } = render(
    <UserHeader
      user={mockUser}
      profileImage="/images/default-avatar.png"
      {...baseProps}
    />
  );

  expect(getByText('John Doe')).toBeInTheDocument();
  expect(getByText('john.doe@example.com')).toBeInTheDocument();
  expect(getByAltText('User Avatar')).toBeInTheDocument();

  // 🔍 Snapshot
  expect(asFragment()).toMatchSnapshot();
});

test('renders default avatar when profileImage is empty', () => {
  const { getByAltText } = render(
    <UserHeader
      user={mockUser}
      profileImage=""
      {...baseProps}
    />
  );

  const avatar = getByAltText('User Avatar');
  expect(avatar).toBeInTheDocument();
  expect(avatar.src).toContain('/images/default-avatar.png');
});

test('renders fallback name when firstName is missing', () => {
  const user = { ...mockUser, firstName: '', lastName: '' };
  const { getByText } = render(
    <UserHeader user={user} profileImage="" {...baseProps} />
  );

  expect(getByText('Анонимный пользователь')).toBeInTheDocument();
});

test('renders fallback email when email is missing', () => {
  const user = { ...mockUser, email: '' };
  const { getByText } = render(
    <UserHeader user={user} profileImage="" {...baseProps} />
  );

  expect(getByText('Email не указан')).toBeInTheDocument();
});

test('renders fallback name and email when both are missing', () => {
  const user = {
    firstName: '',
    lastName: '',
    email: '',
    imageUrl: '',
  };

  const { getByText } = render(
    <UserHeader user={user} profileImage="" {...baseProps} />
  );

  expect(getByText('Анонимный пользователь')).toBeInTheDocument();
  expect(getByText('Email не указан')).toBeInTheDocument();
});