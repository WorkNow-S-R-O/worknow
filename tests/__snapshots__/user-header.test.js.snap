// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UserHeader correctly 1`] = `
<DocumentFragment>
  <div
    class="d-flex flex-column align-items-center mb-4"
  >
    <img
      alt="User Avatar"
      class="rounded-circle mb-3"
      src="/images/default-avatar.png"
      style="width: 100px; height: 100px; object-fit: cover;"
    />
    <div>
      <h2>
        John Doe
      </h2>
      <p
        class="text-muted"
      >
        john.doe@example.com
      </p>
    </div>
  </div>
</DocumentFragment>
`;
